torch/_inductor/codegen/wrapper.py

```diff
@@ -911,28 +920,63 @@ def write_async_compile_wait(self) -> None:
             """
         )
 
+    def write_args(self, input_names: List[str]):
+        lhs = ", ".join(input_names)
+        if len(input_names) == 1:
+            lhs += ","
+        self.prefix.writeline(f"{lhs} = args")
+        self.prefix.writeline("args.clear()")
+
     def write_prefix(self) -> None:
         assert self.launcher_fn_name is not None
         self.write_async_compile_wait()
-        self.prefix.splice(
-            f"""
-            def {self.launcher_fn_name}(args):
-            """
-        )
-        with self.prefix.indent():
+        if (
+            config.graph_partition
+            and not V.graph.cpp_wrapper
+            and not V.graph.aot_mode
+            and not hasattr(self, "subgraph_name")
+        ):
+            self.prefix.splice(
+                """
+                class Runner:
+                    def __init__(self):
+                        self.partitions = []
+
+                    def recursively_apply_fns(self, fns):
```

Comment:
This is a todo for future pr i suppose but: we'll need to figure out what the cudagraph metadata is (static_input) etc for each partition..